### PR TITLE
Fix deadlocks between write transactions and hotbackup, take 2

### PR DESCRIPTION
**Pull request by neunhoef**
Make sure all follower trxs know that they are follower trxs.

Before this patch, follower transactions created from AQL or other
El Cheapo transactions did not know they are follower transactions.
This could cause deadlocks between write transactions and hotbackups.

This is the port to the 3.7.3 branch.

### CODE REVIEW

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/12690/

**Review by jsteemann — APPROVED** [review]

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/12693/

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/12694/

File: tests/js/client/shell/shell-deadlock-trx-hotbackup-cluster.js [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+      if (!isCluster) {
+        return true;
+      }
+
+      wait(3);   // give the cluster some time to settle before we start
+                 // the hotbackups
+
+      let start = time();
+      // This will create lots of hotbackups for approximately 60 seconds
+      db._connection.POST("_admin/execute",`require("@arangodb/cluster.js").createManyBackups(60); return true;`,{"x-arango-async":"store"});
```
Comment (jsteemann) [inline tests/js/client/shell/shell-deadlock-trx-hotbackup-cluster.js:67]:
```suggestion
      arango.POST_RAW("_admin/execute",`require("@arangodb/cluster.js").createManyBackups(60); return true;`,{"x-arango-async":"store"});
```

File: tests/js/client/shell/shell-deadlock-trx-hotbackup-cluster.js [hunk 2]
```diff
@@ -0,0 +1,99 @@
...
+                   RETURN NEW`).toArray();
+        if (i % 50 === 0) {
+          print("Done", i, "write transactions.");
+        }
+        wait(0.01);
+      }
+      let diff = time() - start;
+      print("Done 750 exclusive transactions in", diff, "seconds!");
+      assertTrue(diff < 60, "750 transactions took too long, probably some deadlock with hotbackups");
+      while (time() - start < 65) {
```
Comment (jsteemann) [inline tests/js/client/shell/shell-deadlock-trx-hotbackup-cluster.js:89]: Poll `/_api/job/<id>` until job is finished.

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/12710/

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/12715/

File: tests/js/client/shell/shell-deadlock-trx-hotbackup-cluster.js (hunk 2 above)
Comment (neunhoef) [reply tests/js/client/shell/shell-deadlock-trx-hotbackup-cluster.js]: Done

File: tests/js/client/shell/shell-deadlock-trx-hotbackup-cluster.js (hunk 1 above)
Comment (neunhoef) [reply tests/js/client/shell/shell-deadlock-trx-hotbackup-cluster.js]: Done.

**Comment by neunhoef** [comment]
http://jenkins01.arangodb.biz:8080/view/PR/job/arangodb-matrix-pr/12720/

**Comment by KVS85** [comment]
http://jenkins.arangodb.biz:8080/job/arangodb-matrix-pr/12723/

**Comment by KVS85** [comment]
Tests blue.

**Pull request merged**